controllers/llamastackdistribution_controller.go

```diff
@@ -254,38 +258,123 @@ func (r *LlamaStackDistributionReconciler) reconcileManifestResources(ctx contex
 		return fmt.Errorf("failed to filter manifests: %w", err)
 	}
 
+	// Delete excluded resources that might exist from previous reconciliations
+	if err := r.deleteExcludedResources(ctx, instance, kindsToExclude); err != nil {
+		return fmt.Errorf("failed to delete excluded resources: %w", err)
+	}
+
+	// Apply resources to cluster
 	if err := deploy.ApplyResources(ctx, r.Client, r.Scheme, instance, filteredResMap); err != nil {
 		return fmt.Errorf("failed to apply manifests: %w", err)
 	}
 
 	return nil
 }
 
-// reconcileResources reconciles all resources for the LlamaStackDistribution instance.
-func (r *LlamaStackDistributionReconciler) reconcileResources(ctx context.Context, instance *llamav1alpha1.LlamaStackDistribution) error {
-	// Reconcile ConfigMaps
-	if err := r.reconcileConfigMaps(ctx, instance); err != nil {
-		return err
+// deleteExcludedResources deletes resources that are excluded from the current reconciliation
+// but might exist from previous reconciliations.
+func (r *LlamaStackDistributionReconciler) deleteExcludedResources(ctx context.Context, instance *llamav1alpha1.LlamaStackDistribution, kindsToExclude []string) error {
+	logger := log.FromContext(ctx)
+
+	for _, kind := range kindsToExclude {
```

Comment:
Use `if slices.Contains(kindsToExclude, "NetworkPolicy")` instead of the loop?

Reply:
Done